pinot-core/src/main/java/org/apache/pinot/core/query/pruner/SegmentPruner.java

```diff
@@ -41,9 +43,12 @@ default List<IndexSegment> prune(List<IndexSegment> segments, QueryContext query
       return segments;
     }
     List<IndexSegment> selectedSegments = new ArrayList<>(segments.size());
-    for (IndexSegment segment : segments) {
-      if (!prune(segment, query)) {
-        selectedSegments.add(segment);
+    try (InvocationScope scope = Tracing.getTracer().createScope(getClass())) {
```

Comment:
Suggest moving this trace into the `SegmentPrunerService` because this whole method might be overridden

Reply:
good suggestion